core/providers/utils/makerequest_test.go

```
@@ -101,16 +101,88 @@ func TestMakeRequestWithContext_DeadlineExceededReturnsTimeoutError(t *testing.T
+		t.Fatalf("wait() blocked for %v after context timeout", elapsed)
 	}
 
 	// Now safe to release
 	fasthttp.ReleaseRequest(req)
 	fasthttp.ReleaseResponse(resp)
 }
 
+func TestMakeRequestWithContext_ContextDeadlineExtendsClientReadTimeout(t *testing.T) {
+	// Use a real TCP listener to exercise fasthttp's normal client path. The
+	// upstream accepts the request but deliberately never sends a response.
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)